src/utils/http.ts

```diff
@@ -32,25 +35,41 @@ const convertAxiosError = (e: AxiosError): Error => {
     );
 };
 
+
+/**
+ * Normalizes the total size of a download / upload event, and calls the provided onProgress
+ * callback with loaded and total sizes.
+ *
+ * @param onProgress
+ * @return The handler that wraps `onProgress`.
+ */
+const normalizeTotalSize = (onProgress: ProgressCallback) => ({
+    loaded,
+    total,
+}: AxiosProgressEvent) => {
+    if ("undefined" === typeof total || isNaN(total)) {
+        total = loaded;
+    }
+    onProgress(loaded, total);
+};
+
 /**
  * Downloads (bypassing any caching) a file as a Uint8Array.
  *
  * @param fileUrl
- * @param progressCallback
+ * @param [onProgress]
  * @return The file's content.
  * @throws {Error} if the download fails.
  */
-const getUint8ArrayFrom = async (fileUrl: string, progressCallback: ProgressCallback)
+const getUint8ArrayFrom = async (
+    fileUrl: string,
+    onProgress: ProgressCallback = () => null
```

Comment:
Note now we provide a default value to `onProgress` if it's not provided by the caller.